src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -1,39 +1,67 @@
 package codeu.controller;
 
 import java.io.IOException;
-
+import java.util.List;
+import java.util.UUID;
+import java.time.Instant;
 import codeu.model.data.User;
+import codeu.model.data.Conversation;
+import codeu.model.data.Message;
 import codeu.model.store.basic.UserStore;
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
 import javax.servlet.ServletException;
 import javax.servlet.http.HttpServlet;
-import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
 
-@WebServlet(urlPatterns = "/MyProfile")
 public class ProfileServlet extends HttpServlet {
 
-	private UserStore userStore;
-
-	@Override
-	public void doGet(HttpServletRequest request, HttpServletResponse response) 
-		throws IOException, ServletException {
-			request.getRequestDispatcher("/WEB-INF/view/profile.jsp").forward(request, response);
-		}
-	@Override
-	public void doPost(HttpServletRequest request, HttpServletResponse response)
-		throws IOException, ServletException {
+  /** Store class that gives access to Users. */
+ 	private UserStore userStore;
 
-		// get username
-		String username = (String) request.getSession().getAttribute("user");
+  /** Set up state for handling user requests. */
+  @Override
+  public void init() throws ServletException {
+    super.init();
+    setUserStore(UserStore.getInstance());
+  }
 
-		User user = userStore.getUser(username);
-		// user is not logged in, do not allow access  
-		if (user == null) {
-			response.sendRedirect("/MyProfile");
-			return;
-		}
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
 
+  /**
+   * This function fires when a user navigates to the profiles page. It gets the user's name from
+   * the URL and then forwards to profile.jsp for rendering.
+   */
+ 	@Override
```

Comment:
Looks like there still of tabs going on here.  I'm not sure what IDE you use, but there's likely a way to set it so that when you hit the tab key, it'll put in two spaces instead.